### PR TITLE
Fix false positive in BindingConstantWithoutTypeAlias

### PR DESCRIPTION
**Pull request by ghiculescu**
Fixes https://github.com/Shopify/rubocop-sorbet/issues/37

### CODE REVIEW

File: lib/rubocop/cop/sorbet/binding_constants_without_type_alias.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ module Sorbet
       #   FooOrBar = T.type_alias { T.any(Foo, Bar) }
       class BindingConstantWithoutTypeAlias < RuboCop::Cop::Cop
         def_node_matcher(:binding_unaliased_type?, <<-PATTERN)
-          (casgn _ _ [#not_nil? #not_t_let? #method_needing_aliasing_on_t?])
+          (casgn _ _ [#not_nil? #not_t_let? #not_type_member? #method_needing_aliasing_on_t?])
```
Comment (Morriar) [inline lib/rubocop/cop/sorbet/binding_constants_without_type_alias.rb:20]: Should we also check for `type_template`: https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20Box%0A%20%20extend%20T%3A%3AGeneric%0A%0A%20%20Elem%20%3D%20type_template%0Aend ?

Comment (ghiculescu) [reply lib/rubocop/cop/sorbet/binding_constants_without_type_alias.rb:20]: Probably - I can add that too. I didn't know that was a thing.

Comment (ghiculescu) [reply lib/rubocop/cop/sorbet/binding_constants_without_type_alias.rb]: https://github.com/Shopify/rubocop-sorbet/pull/38/commits/c33c7d92d27c5a992d877e6aead7bcf9c1d06613

**Comment by Morriar** [comment]
Thanks!

**Comment by Morriar** [comment]
It's in the box, thank you both.

**Pull request merged**